subsys/mgmt/hawkbit/hawkbit.c

```diff
@@ -827,6 +857,7 @@ static bool send_request(enum http_method method, enum hawkbit_http_request type
 	hb_context.http_req.header_fields = (const char **)headers;
 #endif
 	hb_context.final_data_received = false;
+	hb_context.status_buffer_size = sizeof(hb_context.status_buffer);
```

Comment:
Good catch! This looks like a potential bug fix and should probably be in another PR, so that we can backport it to earlier releases.

I also wonder if it make sense to remove `status_buffer_size` from the struct and simply use `sizeof(hb_context.status_buffer) - 1` instead of `hb_context.status_buffer_size - 1`?

Reply:
this is now addressed in #70355